internal/abci/abci.go

```diff
@@ -318,6 +317,16 @@ func (a *AbciApp) FinalizeBlock(ctx context.Context, req *abciTypes.RequestFinal
 
 	res.ValidatorUpdates = make([]abciTypes.ValidatorUpdate, len(valUpdates))
 	for i, up := range valUpdates {
+		addr, err := pubkeyToAddr(up.PubKey)
+		if err != nil {
+			return nil, fmt.Errorf("failed to convert pubkey to address: %w", err)
+		}
+		if up.Power == 0 {
+			delete(a.validatorAddressToPubKey, addr)
```

Comment:
Okay we are deleting this from the map. So if a validator leaves and it becomes the proposer in the next block, we would just reject that proposed block.